boltdesignsystem/bolt
# DS-790 Content items deprecation

## Jira

https://pegadigitalit.atlassian.net/browse/DS-790

## Summary

The props `content_items` and `contentItems` were deprecated. The new prop `items` was added.

## Details

The usage of `content_items` and `contentItems` were replaced with `items`. Additionally, objects with image attributes (`src`, `width`, `height`, etc.) were updated with the image set, to pass data the correct/new way, as was said in [this discussion](https://github.com/boltdesignsystem/bolt/pull/2478#discussion_r874124997)

:arrow_down: :arrow_down: :arrow_down: :arrow_down:

**Old usage** :x:
```
{% include '@bolt-components-background/background.twig' with {
    opacity: 'heavy',
    fill: 'gradient',
    content_items: [
      {
        pattern: 'image',
        src: '/images/content/backgrounds/background-tall-4.jpg',
        ...
      },
    ]
} only %}
```

**New usage** :white_check_mark:
```
{% set image %}
  {% include '@bolt-elements-image/image.twig' with {
    background: true,
    attributes: {
      src: '/images/content/backgrounds/background-tall-4.jpg',
      ...
    },
  } only %}
{% endset %}
{% include '@bolt-components-background/background.twig' with {
  opacity: 'heavy',
  fill: 'gradient',
   items: [
    image
  ]
} only %}
```

Updated in places wherever `pattern: 'image'` was in usage:

- Layouts folder
- Components folder
-- Background
-- Band
-- Breadcrumb
-- Card
-- Carousel
-- Modal
-- Sticky
- Pages folder
-- Boc
- Experiments folder
- Archive
-- Academy
-- Content Hub
-- Events
-- Homepage
-- Partners
-- Product

## How to test

Pull The Branch. Check if there are no errors related to the new prop (`items`) usage. Confirm that we can pass an object with image attributes to the `items: [...]` and an image variable.

## Release notes

The props `content_items` and `contentItems` related with the `bolt-background` component are deprecated. Please use the correct way of passing an image to the new prop `items`.
